test.py

```diff
@@ -563,16 +563,65 @@ def run_test(framework, cluster, repo_root):
         logger.info(msg, runtests_script, framework.name)
         raise CommandFailure(cmd_args)
 
+def setup_junit_xml():
+    """Try to download and import junit_xml"""
+    workdir = get_work_dir()
+    lib_dir = os.path.join(workdir, 'lib', 'python')
+    pip_cmd = ['pip', 'install', 'junit-xml', '-t', lib_dir]
```

Comment:
Are you objecting to the pattern of a process downloading and loading a module itself, or just looking for different module organization?

I have no problem with moving this single function into another module, but there wont' be any meaningful encapsulation or interface, partly because junit_xml is pretty awful and doesn't offer interfaces for sensibly building out its state iteratively.   (Also solvable but there's some definite diminishing returns here.) 

If it's a goal always using subprocesses for loading modules dynamically, that's definitely not something I want.  I want exception outputs that give clear traces of failures.  I have many years of experience of being able to produce fixes for very specific edge-case problem from a single error log by following good-stewardship patterns allowing quality exception output.  We are currently not in that position, and many failures are question marks for precisely this reason.